url2b4a2d655b_Yehonal_unisa_prog2_jdm_tgz-pJ8-com_jdeals_tests_TestUserBuyingJ8/src/com/jdeals/libs/MyDate.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2007 - 2014 Hyperweb2 All rights reserved.
+ * GNU General Public License version 3; see www.hyperweb2.com/terms/
+ */
+package com.jdeals.libs;
+
+import java.io.Serializable;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Scanner;
+
+import javax.swing.text.MaskFormatter;
+
+/**
+ * This class would simplify date conversion/parsing and adds different tools
+ * for date operations.
+ *
+ * @author Giuseppe Ronca
+ */
+public class MyDate implements Serializable, Cloneable {
+
+    /**
+     * The Constant defFormat.
+     */
+    public static final String defFormat = "dd/mm/yyyy";
+
+    /**
+     * The Constant defMask.
+     */
+    public static final String defMask = "##/##/####";
+
+    /**
+     * The Enum DiffType.
+     */
+    public enum DiffType {
+
+        /**
+         * The days.
+         */
+        DAYS(1000.0 * 24.0 * 60.0 * 60.0);
+
+        /**
+         * The val.
+         */
+        private double val;
+
+        /**
+         * Instantiates a new diff type.
+         *
+         * @param val the val
+         */
+        private DiffType(double val) {
+            this.val = val;
+        }
+
+        /**
+         * Gets the val.
+         *
+         * @return the val
+         */
+        public double getVal() {
+            return this.val;
+        }
+    }
+
+    /**
+     * The format.
+     */
+    private String format;
+
+    /**
+     * The calendar.
+     */
+    private Calendar calendar;
+
+    /**
+     * Instantiates a new my date.
+     *
+     * @param format the format
+     * @param date the date
+     */
+    public MyDate(String format, Calendar date) {
+        this.format = format;
+        this.calendar = date;
+    }
+
+    /**
+     * Instantiates a new my date.
+     */
+    public MyDate() {
+        this(MyDate.defFormat, Calendar.getInstance());
+    }
+
+    /**
+     * From string.
+     *
+     * @param date the date
+     * @return the my date
+     * @throws Exception the exception
+     */
+    public MyDate fromString(String date) throws Exception {
+        // Date now = new Date();
+        DateFormat df = new SimpleDateFormat(this.format);
+        df.setLenient(false);
+
+        try {
+            this.calendar.setTime(df.parse(date));
+        } catch (ParseException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)